src/enterprise-edge/azext_enterprise_edge/custom.py

```diff
@@ -52,18 +52,25 @@ def get(cls, cmd, resource_group, name):
         return cls._request(cmd, resource_group, name)
 
     @classmethod
-    def _validate_cdn_provider_registered(cls, cmd):
+    def _register_cdn_provider(cls, cmd):
         management_hostname = cmd.cli_ctx.cloud.endpoints.resource_manager
         api_version = "2021-04-01"
         sub_id = get_subscription_id(cmd.cli_ctx)
-        url_fmt = "{}/subscriptions/{}/providers/Microsoft.CDN?api-version={}"
```

Comment:
any reason we are not using Python SDK function directly here?

Reply:
Do you know the SDK function off-hand? I've been searching through the SDK code and the SDK browser, but I have not found it